apis/Google.Devtools.AspNet/Google.Devtools.AspNet.IntegrationTests/CloudTrace/CloudTraceIntregrationTest.cs

```diff
@@ -0,0 +1,288 @@
+
+using Google.Cloud.Trace.V1;
+using Google.Protobuf.Collections;
+using Google.Protobuf.WellKnownTypes;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+using Trace = Google.Cloud.Trace.V1.Trace;
+
+namespace Google.Devtools.AspNet.IntegrationTests
+{
+    public class CloudTraceIntregrationTest
+    {
+        /// <summary>Total time to spend sleeping when looking for a trace.</summary>
+        private const int TimeoutMilliseconds = 10000; // 10 seconds.
+
+        /// <summary>Time to sleep between checks for a trace.</summary>
+        private const int SleepIntervalMilliseconds = 1000; // 1 second.
+
+        /// <summary>Project id to run the test on.</summary>
+        private readonly string ProjectId;
+
+        /// <summary>Unique id of the test.</summary>
+        private readonly string TestId;
+
+        /// <summary>Test start time to allow for easier querying of traces.</summary>
+        private readonly Timestamp StartTime;
+
+        /// <summary>Using the underlying grpc client so we can filter.</summary>
+        private readonly TraceService.TraceServiceClient GrpcClient;
+
+        public CloudTraceIntregrationTest()
+        {
+            ProjectId = Utils.GetProjectIdFromEnvironment();
+            TestId = Utils.GetTestId();
+            StartTime = Timestamp.FromDateTime(DateTime.Now.ToUniversalTime());
+            GrpcClient = TraceServiceClient.Create().GrpcClient;
+        }
+
+        private GrpcTraceConsumer CreateGrpcTraceConsumer()
+        {
+            TraceServiceClient client = TraceServiceClient.Create();
+            return new GrpcTraceConsumer(Task.FromResult(client));
+        }
+
+        private SimpleManagedTracer CreateSimpleManagedTracer(ITraceConsumer consumer)
+        {
+            TraceIdFactory traceIdFactory = TraceIdFactory.Create();
+            Trace trace = new Trace()
+            {
+                ProjectId = ProjectId,
+                TraceId = traceIdFactory.NextId()
+            };
+            return SimpleManagedTracer.Create(consumer, trace, null);
+        }
+
+        /// <summary>
+        /// Gets a trace that contains a span with the given name.
+        /// </summary>
+        private async Task<Trace> GetTrace(string spanName)
+        {
+            int sleepTime = 0;
+            while (sleepTime < TimeoutMilliseconds)
+            {
+                sleepTime += SleepIntervalMilliseconds;
+                Thread.Sleep(SleepIntervalMilliseconds);
+
+                ListTracesRequest request = new ListTracesRequest
+                {
+                    ProjectId = ProjectId,
+                    StartTime = StartTime,
+                    View = ListTracesRequest.Types.ViewType.Complete
+                };
+                ListTracesResponse response = await GrpcClient.ListTracesAsync(request);
```

Comment:
I think I'm missing how to do this.  All I see is the project id as a param and no other options.